lib/content-services/src/lib/security/services/security-controls-groups-marks-security.service.spec.ts

```diff
@@ -0,0 +1,270 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TestBed } from '@angular/core/testing';
+import { CoreTestingModule } from '@alfresco/adf-core';
+import { SecurityControlsService } from './security-controls-groups-marks-security.service';
+import {
+    fakeGroupsApiResponse,
+    createNewSecurityGroupMock
+} from './mock/security-groups.mock';
+import {
+    fakeMarksApiResponse,
+    createNewSecurityMarkMock
+} from './mock/security-marks.mock';
+import { of } from 'rxjs';
+import {
+    SecurityGroupBody,
+    SecurityGroupEntry,
+    SecurityMarkBody,
+    SecurityMarkEntry
+} from '@alfresco/js-api';
+
+describe('SecurityControlsService', () => {
+    let service: SecurityControlsService;
+    let securityGroupId;
+    let securityMarkId;
+    const securityGroupBody: SecurityGroupBody = {
+        groupName: 'TestGroup',
+        groupType: 'HIERARCHICAL'
+    };
+    const securityMarkBody: SecurityMarkBody = {
+        name: 'securityMark1'
+    };
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [CoreTestingModule]
+        });
+
+        service = TestBed.inject(SecurityControlsService);
+    });
+
+    it('should be able to get the list of Security groups', async () => {
+        const getGroupSpy = spyOn(
+            service.groupsApi,
+            'getSecurityGroups'
+        ).and.returnValue(Promise.resolve(fakeGroupsApiResponse));
+        const groupPromise = service.getSecurityGroup( 0, 5, 'inUse');
+        const group = await groupPromise;
+        expect(getGroupSpy).toHaveBeenCalledWith({
+            skipCount: 0,
+            maxItems: 5,
+            include: 'inUse'
+        });
+
+        expect(group.pagination.skipCount).toBe(0);
+        expect(group.pagination.maxItems).toBe(5);
+        expect(group.entries[0].id).toBe('classification');
+        expect(group.entries[0].groupName).toBe('Classification');
+        expect(group.entries[0].groupType).toBe('HIERARCHICAL');
+
+        expect(group.entries[1].id).toBe(
+            'd2b11d9f-2707-439f-a7c6-e7872f395553'
+        );
+        expect(group.entries[1].groupName).toBe('SG1');
+        expect(group.entries[1].groupType).toBe('USER_REQUIRES_ALL');
+
+        expect(group.entries[2].id).toBe(
+            '1b77a32d-6b8b-4a37-b195-7f2ff2fe4ed3'
+        );
+        expect(group.entries[2].groupName).toBe('SG2');
+        expect(group.entries[2].groupType).toBe('USER_REQUIRES_ALL');
+
+        expect(group.entries[3].id).toBe(
+            '709791f8-22dc-428a-82dd-daf3e1aa8a60'
+        );
+        expect(group.entries[3].groupName).toBe('SG3');
+        expect(group.entries[3].groupType).toBe('USER_REQUIRES_ALL');
+    });
+
+    it('should create new security group', async () => {
+        spyOn(service.groupsApi, 'createSecurityGroup').and.returnValue(
+            // @ts-ignore
```

Comment:
You don't need this probably. You are getting type issue because createSecurityGroup returns Promise, not Observable. Here you use "of" which creates observable. Replace of with for example Promise.resolve(...)

Reply:
resolved, ty